src/clients/python/experimental_api_v2/examples/simple_http_v2_class_client.py

```diff
@@ -0,0 +1,269 @@
+            (input_config['format'] != "FORMAT_NHWC")):
+        raise Exception("unexpected input format " + input_config['format']
+                        + ", expecting FORMAT_NCHW or FORMAT_NHWC")
+
+    if input_config['format'] == "FORMAT_NHWC":
+        h = input_metadata['shape'][0]
+        w = input_metadata['shape'][1]
+        c = input_metadata['shape'][2]
+    else:
+        c = input_metadata['shape'][0]
+        h = input_metadata['shape'][1]
+        w = input_metadata['shape'][2]
+
+    return (input_metadata['name'], output_metadata['name'], c, h, w,
+            "FORMAT_NHWC", input_metadata['datatype'])
+
+
+def preprocess(img, format, dtype, c, h, w, scaling):
+    """
+    Pre-process an image to meet the size, type and format
+    requirements specified by the parameters.
+    """
+    # np.set_printoptions(threshold='nan')
+
+    if c == 1:
+        sample_img = img.convert('L')
+    else:
+        sample_img = img.convert('RGB')
+
+    resized_img = sample_img.resize((w, h), Image.BILINEAR)
+    resized = np.array(resized_img)
+    if resized.ndim == 2:
+        resized = resized[:, :, np.newaxis]
+
+    npdtype = triton_to_np_dtype(dtype)
+    typed = resized.astype(npdtype)
+
+    if scaling == 'INCEPTION':
+        scaled = (typed / 128) - 1
+    elif scaling == 'VGG':
+        if c == 1:
+            scaled = typed - np.asarray((128,), dtype=npdtype)
+        else:
+            scaled = typed - np.asarray((123, 117, 104), dtype=npdtype)
+    else:
+        scaled = typed
+
+    # Swap to CHW if necessary
+    if format == "FORMAT_NCHW":
+        ordered = np.transpose(scaled, (2, 0, 1))
+    else:
+        ordered = scaled
+
+    # Channels are in RGB order. Currently model configuration data
+    # doesn't provide any information as to other channel orderings
+    # (like BGR) so we just assume RGB.
+    return ordered
+
+
+def postprocess(results, output_name, batch_size):
+    """
+    Post-process results to show classifications.
+    """
+
+    output_array = results.as_numpy(output_name)
+    if not "VULTURE" in str(output_array[0][0]):
+        raise Exception(
+            "expected VULTURE as the first result, instead received: {}".format(
+                output_array[0][0]))
+
+
+def requestGenerator(input_name, output_name, c, h, w, format, dtype, FLAGS):
+    # Preprocess image into input data according to model requirements
+    image_data = None
+    with Image.open(FLAGS.image_filename) as img:
+        image_data = preprocess(img, format, dtype, c, h, w, FLAGS.scaling)
+
+    repeated_image_data = [image_data for _ in range(FLAGS.batch_size)]
+    batched_image_data = np.stack(repeated_image_data, axis=0)
+
+    # Set the input data
+    inputs = []
+    inputs.append(httpclient.InferInput(input_name, batched_image_data.shape, dtype))
+    inputs[0].set_data_from_numpy(batched_image_data)
+
+    outputs = []
+    outputs.append(httpclient.InferOutput(output_name, class_count=FLAGS.classes))
```

Comment:
You would have to use "binary_data=False" here and any 'set_data_from_numpy' calls. Have a look at #1253. Once DLIS-1204 is fixed all the example clients can enable binary data.

Reply:
Done